File: Java/Medium/Accepted/finding-the-users-active-minutes_7.java
class Solution {
    public int[] findingUsersActiveMinutes(int[][] logs, int k) {
        
        int[] result = new int[k];
        Map<Integer, Set<Integer>> map = new HashMap<>();
        for (int[] log : logs) {
            int user = log[0];
            int minute = log[1];
            if (!map.containsKey(user)) {
                map.put(user, new HashSet<>());
            }
            map.get(user).add(minute);
        }
        for (Integer user : map.keySet()) {
            int size = map.get(user).size();
            result[size - 1]++;
        }
        return result;
    }
}

File: C++/Medium/Accepted/finding-the-users-active-minutes_7.cpp
class Solution {
public:
    vector<int> findingUsersActiveMinutes(vector<vector<int>>& logs, int k) {
        
        vector<int> ans(k, 0);
        unordered_map<int, unordered_set<int>> uam;
        for (auto& log : logs) {
            uam[log[0]].insert(log[1]);
        }
        for (auto& [_, v] : uam) {
            ans[v.size() - 1]++;
        }
        return ans;
    }
};